### DISCUSSION TITLE
Need information about Gotham

### DISCUSSION THREAD

**Discussion by JeromeBourgeois** [gotham-rs/gotham#590] [General]
Hi,

 What are the motivations for creation and the main difference between Gotham and Actix ?
 
 Thanks.

**Comment by tanriol** [comment]
This is not a full answer, but I'd like to note that Gotham is a few months older than Actix-web. When 0.0.1 and 0.1.0 of Gotham were published, there was no `axum`, `warp`, `tide`, `actix-web`. There was `rocket`, but it used to require Nightly. That was the decline of `iron` and `nickel`, and possibly other frameworks I've forgotten... so the "motivation for creation" was likely to have a web framework for stable Rust with some specific features over the ones that existed at that time (or just a new and live one in case these were already stuck in their development).

**Reply by msrd0** [reply]
To add to that, I dug up the initial announcement for gotham: https://users.rust-lang.org/t/announcing-gotham/12307. At the time there weren't many web frameworks around targeting stable Rust, which has changed in the years since.

**Reply by tanriol** [reply]
Yeah, looks like I've forgotten that Gotham was one of the first async web frameworks for Rust. The preexisting ones were mostly or all non-async, and migration to async was a major stumbling point for them.

**Comment by JeromeBourgeois** [comment]
Oh ok, thanks for all that information, I have the Rust Web history timeline ;)
